File: BINARYIO - Binary Input and Output/Binary Input and Output.py
# Project name : SPOJ: BINARYIO - Binary Input and Output
# Author       : Wojciech Raszka
# Date created : 2019-03-10
# Description  :
# Status       : Accepted (23379972)
# Comment      :

import sys
import struct
from math import log

for block in iter(lambda: sys.stdin.buffer.read(8), ""):
    if block:
        x = log(struct.unpack("d", block)[0])
        sys.stdout.buffer.write(struct.pack("d", x))
    else:
        break

File: BINARYIO - Binary Input and Output/Binary Input and Output.cpp
/*
* Project name : SPOJ: BINARYIO - Binary Input and Output
* Author       : Wojciech Raszka
* Date created : 2019-03-10
* Description  :
* Status       : Accepted (23379792)
* Comment      :
*/

#include <stdio.h>
#include <math.h>

int main(){
  double x;

  while (fread(&x, sizeof(x), 1, stdin) == 1){
    x = log(x);
    fwrite(&x, sizeof(x), 1, stdout);
  }

  return 0;
}
